CrowsZA/functions/fn_surrenderChance.sqf

```diff
@@ -0,0 +1,185 @@
+];
+
+if(isNull _unit) exitWith { false };
+
+if(_unit getVariable ["crowsza_surrender_chance_applied", false]) exitWith { ["Surrender chance already applied to this unit"] call crowsZA_fnc_showHint; false };
+
+
+// TODO: add to CZA draw handler so zeuses can see which units have had this applied (similar to Remote Control)
+_unit setVariable ["crowsza_surrender_chance_applied", true, true];
+
+
+if(_holdFire) then {
+	// Set the combat mode of the whole group, to avoid the entire group
+	// opening fire when one "snaps"
+	_unit setCombatMode "BLUE";
+};
+
+
+private _trigger = createTrigger ["EmptyDetector", getPos _unit, false];
+_trigger setTriggerArea [_confrontationRadius, _confrontationRadius, 0, false];
+_trigger setTriggerActivation ["ANYPLAYER", "PRESENT", false];
+
+_trigger setVariable ["_unit", _unit];
+_unit setVariable ["_trigger", _trigger];
+_unit setVariable ["_hesitation", _hesitation];
+_unit setVariable ["_surrenderChance", _surrenderChance];
+
+_trigger attachTo [_unit];
+
+
+// TODO: Optionally ensure that the player pointing a weapon is considered HOSTILE to the units side
+// (don't mandate - could be useful to encourage friendly units, e.g. CIVs, to surrender too)
+
+private _condition = "
+if(!this) exitWith { false };
+private _unit = thisTrigger getVariable ""_unit"";
+{
+	[_unit, {
+		missionNamespace setVariable [
+			""crowsza_surrender_chance_"" + str _this,
+			(toUpperANSI cameraView == ""GUNNER"") && (cursorObject == _this),
+			remoteExecutedOwner
+		];
+	}] remoteExecCall [""call"", _x];
+
+	if((missionNamespace getVariable (""crowsza_surrender_chance_"" + str _unit))) exitWith {true};
+} forEach thisList;
+(missionNamespace getVariable (""crowsza_surrender_chance_"" + str _unit))
+";
+
+
+private _activation = "
+private _unit = thisTrigger getVariable ""_unit"";
+private _surrenderChance = _unit getVariable ""_surrenderChance"";
+private _hesitation = _unit getVariable ""_hesitation"";
+
+missionNamespace setVariable [(""crowsza_surrender_chance_"" + str _unit), nil];
+_unit setVariable [""crowsza_surrender_chance_applied"", nil, true];
+deletevehicle thisTrigger;
+
+[_unit, _surrenderChance, _hesitation, thisTrigger] spawn {
+
+	params [""_unit"", ""_surrenderChance"", ""_hesitation""];
+
+	sleep ([random (0.5 + _hesitation), 0.5, _hesitation] call BIS_fnc_clamp);
+
+
+	if(random 1 < _surrenderChance) then {
+
+		if (crowsZA_common_aceModLoaded) then {
+			[""ace_captives_setSurrendered"",[_unit,true]] call CBA_fnc_globalEvent;
+		} else {
+			_unit action [""surrender"", _unit];
+		};
+		
+		_weapon = currentWeapon _unit;
+		_unit removeWeapon (currentWeapon _unit); 
+		_weaponHolder = ""WeaponHolderSimulated"" createVehicle [0,0,0]; 
+		_weaponHolder addWeaponCargoGlobal [_weapon,1]; 
+		_weaponHolder setPos (_unit modelToWorld [0,.2,1.2]); 
+		_weaponHolder disableCollisionWith _unit; 
+		_dir = random(360); 
+		_speed = 1; 
+		_weaponHolder setVelocity [_speed * sin(_dir), _speed * cos(_dir),4];
```

Comment:
This exact code is repeated down in event-handler. Not sure if possible to move this to its own public function like 
```sqf
crowsza_surrender_chance_fnc_surrender = {
    the function code...
};
```

Then just call it from both activation and event-handler. You can put it at top of this file like I showed, or make it into a different file 
 in CfgFunctions and call that. I just think it would be better to have one place with the effect that gets called. Avoids multiple changes for identical code, and makes it easier to change in future. 

If you think that makes sense?

Reply:
Yep, that was me being lazy - have moved it out to a separate function now!

One strange issue - when the surrender code fires from the EH, it (sometimes) causes a "No entry bin\config.bin/CfgWeapons" pop-up; any ideas what could be causing that?